### PR TITLE
Integration of the LDAP identity manager

### PR DESCRIPTION
**Pull request by Paannda**
Continue of work on [PR #3805](https://github.com/Ombi-app/Ombi/pull/3805)
Based on the work from @dpraul

After a long time, I'm back to work on this feature. This is sill a WIP, and needs some cleanup to do. Most part is code from [dpraul](https://github.com/dpraul), rebased on the latest develop branch.
I know there is still some modification to do, but I'm more than happy to already receive feedback, and test from users

### CODE REVIEW

File: src/Ombi.Api 2.Service/AppVeyorApi.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
+﻿using System;
```
Comment (tidusjar) [inline src/Ombi.Api 2.Service/AppVeyorApi.cs:1]: Any reason why it seems you can copied the folder?

File: src/Ombi.Api 3.Service/AppVeyorApi.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
+﻿using System;
```
Comment (tidusjar) [inline src/Ombi.Api 3.Service/AppVeyorApi.cs:1]: Here's a 3rd copy :D

File: src/Ombi.Core/Authentication/LdapUserManager 2.cs [hunk 3]
```diff
@@ -0,0 +1,168 @@
+﻿using System;
```
Comment (tidusjar) [inline src/Ombi.Core/Authentication/LdapUserManager 2.cs:1]: We have two files `LdapUserManager` and `LdapUserManager 2`?

File: src/Ombi.Core/Authentication/OmbiUserManager.cs [hunk 4]
```diff
@@ -61,14 +64,46 @@ public class OmbiUserManager : UserManager<OmbiUser>
...
         private readonly IPlexApi _plexApi;
         private readonly IEmbyApiFactory _embyApi;
         private readonly IJellyfinApiFactory _jellyfinApi;
+        private readonly ILdapUserManager _ldapUserManager;
         private readonly ISettingsService<EmbySettings> _embySettings;
         private readonly ISettingsService<JellyfinSettings> _jellyfinSettings;
         private readonly ISettingsService<AuthenticationSettings> _authSettings;
+        private readonly ISettingsService<UserManagementSettings> _userManagementSettings;
+
+        public async Task<OmbiUser> FindUser(string userName)
```
Comment (tidusjar) [inline src/Ombi.Core/Authentication/OmbiUserManager.cs:80]: I'm not sure about this method, It has mixed responsibilities, and it's not just FindUser. It also could potentially create the user.

File: src/Ombi.Core/Engine/Demo/DemoMovieSearchEngine.cs [hunk 5]
```diff
@@ -50,7 +50,7 @@ public async Task<IEnumerable<SearchMovieViewModel>> Search(string search)
             return null;
         }
 
-        new public async Task<IEnumerable<SearchMovieViewModel>> NowPlayingMovies()
+        public async Task<IEnumerable<SearchMovieViewModel>> NowPlayingMovies()
```
Comment (tidusjar) [inline src/Ombi.Core/Engine/Demo/DemoMovieSearchEngine.cs:53]: these changes seem unrelated?

File: src/Ombi.Core/Engine/RecentlyAddedEngine.cs [hunk 6]
```diff
@@ -13,62 +13,53 @@ namespace Ombi.Core.Engine
...
         {
             _plex = plex;
             _emby = emby;
-            _jellyfin = jellyfin;
             _recentlyAddedLog = recentlyAdded;
         }
 
         private readonly IPlexContentRepository _plex;
         private readonly IEmbyContentRepository _emby;
-        private readonly IJellyfinContentRepository _jellyfin;
```
Comment (tidusjar) [inline src/Ombi.Core/Engine/RecentlyAddedEngine.cs:26]: Why have you removed jellyfin from this file? Bad merge?

File: src/Ombi.Core/Engine/V2/CalendarEngine.cs [hunk 7]
```diff
@@ -47,7 +47,7 @@ public Task<List<CalendarViewModel>> GetCalendarData()
                         new ExtraParams
                         {
                             Overview = e.Season?.ChildRequest?.ParentRequest?.Overview ?? string.Empty,
-                            ProviderId = e.Season?.ChildRequest?.ParentRequest?.ExternalProviderId ?? 0,
```
Comment (tidusjar) [inline src/Ombi.Core/Engine/V2/CalendarEngine.cs:50]: These changes seem unrelated, and it should be using the `ExternalProviderId`

**Review by tidusjar — CHANGES_REQUESTED** [review]
It seems like there are some merge issues with the PR (I appreciate it's in a draft, was just going to have an initial look). But great work so far!

File: src/Ombi.Api 2.Service/AppVeyorApi.cs (hunk 1 above)
Comment (Paannda) [reply src/Ombi.Api 2.Service/AppVeyorApi.cs]: Seems like a bad manipulation on my side... Deleted the whole folder

File: src/Ombi.Api 3.Service/AppVeyorApi.cs (hunk 2 above)
Comment (Paannda) [reply src/Ombi.Api 3.Service/AppVeyorApi.cs]: Deleted too

File: src/Ombi.Core/Authentication/LdapUserManager 2.cs (hunk 3 above)
Comment (Paannda) [reply src/Ombi.Core/Authentication/LdapUserManager 2.cs]: Same bad manipulation. Deleted

File: src/Ombi.Core/Engine/Demo/DemoMovieSearchEngine.cs (hunk 5 above)
Comment (Paannda) [reply src/Ombi.Core/Engine/Demo/DemoMovieSearchEngine.cs:53]: Indeed, bad rebase, not sure how I ended up modifying that file...

File: src/Ombi.Core/Engine/V2/CalendarEngine.cs (hunk 7 above)
Comment (Paannda) [reply src/Ombi.Core/Engine/V2/CalendarEngine.cs]: Dev version put back in place

**Comment by Paannda** [comment]
Thanks a lot for your feedback, good that everything was caught early. Not sure how I managed to do that, but still working on solving all the issues

**Comment by Scot-Survivor** [comment]
Any updates on this PR?
